### DISCUSSION TITLE
[Feature] Upload to External Library(EL) on mobile while keeping EL folder struct in sync with album struct

### DISCUSSION THREAD

**Discussion by Ender-Wang** [immich-app/immich#14060] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

- Intention: I have always sorted my images into folders on a local disk, and now I wanna move to Immich, I know I can treat it as an External Library, which will not touch my file structure and file names, but I also wanna upload images on my phone to Immich with the same folder/album name. Such as I have a folder called "pets" on my external library, and then all images in that folder will be uploaded to an album "pets" on Immich, right? Then on my phone, I also have a folder "pets" with some images and I add this folder as one of backup Albums on my phone, would those images be uploaded to my external library directory if I set my upload_location to the same directory as the external library directory?

- 	I saw [a discussion on reddit](https://www.reddit.com/r/immich/comments/1f4scji/comment/lkqwyiz/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) that says Immich treats files as "duplicates" differently between uploaded and external library. That file hashes uploaded to immich via the Upload func are calculated based on the actual file, but hashes for files on external library is based on the file path instead. Is it still the case?

- There's a Storage Template func, which changes file names and stores them in a customized way, it would be nice if we can store the uploaded images from phone to our external library with the same file structure such as: I put a new image in the "pets" folder on my phone, then it's uploaded to the "pets" Album on immich, and immich store this image in the external directory folder called "pets".

- The reason behind this feature request is I wanna keep my folder structure synced with immich album structures, so I can set up a script to backup my external library periodically in case Immich crashes or sth happens to my server which causes Immich db corruption etc.. Then I always have one or more full copies of my external library somewhere else that keeps all my images in an organized way that is recognizable by me, even if immich crashes I can set up a new Immich server and let it syncs folders to albums again. There's a workaround to my request, which is to copy the images on my phone to the desired folder location on the external library and let Immich to sync those new images to their albums, but this does not come out in handy, it would be perfect if we can just sync from phone to external lib while maintaining the folder=album structure.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
This is a dupe/combination of several existing feature requests. Note that uploading to external libraries is not supported and not planned.

**Reply by Ender-Wang** [reply]
Thanks for the clarification. Does Immich still calculate the hash of images on external lib by their path? So the same image file uploaded to Immich will not be treated as a duplicate even if it's the same file on the external library?

**Reply by schuhbacca** [reply]
It does yes.
